### ISSUE TITLE
marshal of interface leads memory corruption; "runtime: name offset out of range"

### ISSUE THREAD

**Issue by mewmew** [goccy/go-json#281]
First off, thanks for working on goccy/go-json! It is really an incredible library for improving the performance of the standard library JSON package.

After some extensive use, I ran into an issue which resulted in a runtime panic. Preferably, this would lead to an error, e.g. "unable to marshal interface type Foo" or something.

Reproducible test case for goccy/go-json@v0.7.7.

Running the program below results in the following runtime panic (probably due to memory corruption based on [similar reported issues in Go](https://github.com/golang/go/issues?q=%22name+offset+out+of+range%22)).

```
runtime: nameOff 0x50e7a0 out of range 0x502000 - 0x559eb2
fatal error: runtime: name offset out of range

goroutine 1 [running]:
runtime.throw({0x52592f, 0x50f9a0})
	go/src/runtime/panic.go:1198 +0x71 fp=0xc0001238b8 sp=0xc000123888 pc=0x430cd1
runtime.resolveNameOff(0xc000123940, 0x50e7a0)
	go/src/runtime/type.go:195 +0x265 fp=0xc000123910 sp=0xc0001238b8 pc=0x454365
reflect.resolveNameOff(0x1, 0x555cc0)
	go/src/runtime/runtime1.go:496 +0x19 fp=0xc000123930 sp=0xc000123910 pc=0x459259
reflect.(*rtype).nameOff(...)
	go/src/reflect/type.go:725
reflect.(*rtype).String(0x555cc0)
	go/src/reflect/type.go:795 +0x25 fp=0xc000123950 sp=0xc000123930 pc=0x472885
fmt.(*pp).handleMethods(0xc00006c8f0, 0x60d860)
	go/src/fmt/print.go:626 +0x383 fp=0xc000123c00 sp=0xc000123950 pc=0x48f803
fmt.(*pp).printArg(0xc00006c8f0, {0x51b1c0, 0x555cc0}, 0x73)
	go/src/fmt/print.go:709 +0x693 fp=0xc000123ca0 sp=0xc000123c00 pc=0x4903b3
fmt.(*pp).doPrintf(0xc00006c8f0, {0x5233b6, 0x1a}, {0xc000123e18, 0x40920d, 0x0})
	go/src/fmt/print.go:1026 +0x288 fp=0xc000123d98 sp=0xc000123ca0 pc=0x492c68
fmt.Sprintf({0x5233b6, 0x1a}, {0xc000123e18, 0x1, 0x1})
	go/src/fmt/print.go:219 +0x59 fp=0xc000123df0 sp=0xc000123d98 pc=0x48d2f9
github.com/goccy/go-json/internal/errors.(*UnsupportedTypeError).Error(0x29)
	goget/pkg/mod/github.com/goccy/go-json@v0.7.7/internal/errors/error.go:100 +0x52 fp=0xc000123e38 sp=0xc000123df0 pc=0x4946d2
runtime.preprintpanics(0xc00005e050)
	go/src/runtime/panic.go:701 +0x7f fp=0xc000123e98 sp=0xc000123e38 pc=0x42f55f
panic({0x50d660, 0xc0000102a0})
	go/src/runtime/panic.go:1145 +0x39a fp=0xc000123f58 sp=0xc000123e98 pc=0x43085a
```

Playground link: https://play.golang.org/p/5S_Em_7IkcA
```go
package main

import json "github.com/goccy/go-json"

type I interface {
	M()
}

func main() {
	data := []I{
		T{},
	}
	if _, err := json.Marshal(data); err != nil {
		panic(err)
	}
}

type T struct{}

func (T) M() {
}
```
Cheers,
Robin

**Comment by goccy** [comment]
Thank you for the reporting ! I investigated this bug and I understood root cause .
I will fix this issue as soon as possible !

**Comment by mewmew** [comment]
@goccy thanks for investigating and finding the root cause! It is an impressive achievement to reach the performance improvement that `go-json` delivers while keeping the API compatible with `encoding/json`.

Wish you all the best and happy coding!

Cheers,
Robin

**Comment by goccy** [comment]
@mewmew 
Thank you ! :)
I resolved this bug with #284 and I will release this fixing as v0.7.8 .
